identity/did/bootstrapper.go

```diff
@@ -0,0 +1,147 @@
+package did
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/config/configstore"
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/savaki/jq"
+)
+
+// Bootstrapper implements bootstrap.Bootstrapper.
+type Bootstrapper struct{}
+
+// BootstrappedDIDService stores the id of the bootstrapper
+const BootstrappedDIDService string = "BootstrappedDIDService"
+
+var smartContractAddresses *config.SmartContractAddresses
+
+// Bootstrap initializes the factory contract
+func (*Bootstrapper) Bootstrap(context map[string]interface{}) error {
+	cfg, err := configstore.RetrieveConfig(false, context)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := context[ethereum.BootstrappedEthereumClient]; !ok {
+		return errors.New("ethereum client hasn't been initialized")
+	}
+	client := context[ethereum.BootstrappedEthereumClient].(ethereum.Client)
+
+	// TODO line will be removed after migration
+	migrateNewIdentityContracts()
+
+	factoryContract, err := bindFactory(getFactoryAddress(), client)
+	if err != nil {
+		return err
+	}
+
+	service := NewService(cfg, factoryContract, client)
+
+	context[BootstrappedDIDService] = service
+
+	return nil
+}
+
+func bindFactory(factoryAddress common.Address, client ethereum.Client) (*FactoryContract, error) {
+	return NewFactoryContract(factoryAddress, client.GetEthClient())
+}
+
+func getFactoryAddress() common.Address {
+	return common.HexToAddress(smartContractAddresses.IdentityFactoryAddr)
+
+}
+
+// Note: this block will be removed after the identity migration is done
+// currently we are using two versions of centrifuge contracts to not break the compatiblitiy
+// ---------------------------------------------------------------------------------------------------------------------
+func migrateNewIdentityContracts() {
+	runNewSmartContractMigrations()
+	smartContractAddresses = getSmartContractAddresses()
+
+}
+
+// RunNewSmartContractMigrations migrates smart contracts to localgeth
+// TODO: func will be removed after migration
+func runNewSmartContractMigrations() {
+
+	gp := os.Getenv("GOPATH")
+	projDir := path.Join(gp, "src", "github.com", "centrifuge", "go-centrifuge")
+
+	smartContractDir := path.Join(projDir, "vendor", "github.com", "manuelpolzhofer", "centrifuge-ethereum-contracts")
+	smartContractDirStandard := path.Join(projDir, "vendor", "github.com", "centrifuge", "centrifuge-ethereum-contracts")
+
+	os.Setenv("CENT_ETHEREUM_CONTRACTS_DIR", smartContractDir)
+
+	migrationScript := path.Join(projDir, "build", "scripts", "migrate.sh")
+	_, err := exec.Command(migrationScript, projDir).Output()
```

Comment:
Can we pass it instead as part of the envs of the `exec.Command`? Instead of modifying the global env. Look at: https://github.com/centrifuge/go-centrifuge/blob/develop/cmd/centrifuge_cmd_test.go#L29-L33

Reply:
As discussed. The registry is not existing any more and the sleep func will be removed as soon as #663 is merged.

Reply:
I would need to modify the testingutils package to make it work. 

I will keep it as it is now, because the code will be removed anyway